src/gen_rpc_acceptor.erl

```diff
@@ -197,7 +197,10 @@ call_worker(Parent, WorkerPid, Ref, M, F, A) ->
     % and manifest as timeout. Wrap inside anonymous function with catch
     % will crash the worker quickly not manifest as a timeout.
     % See call_MFA_undef test.
-    % Fun = fun() -> catch erlang:apply(M, F, A) end,
-    PacketBin = erlang:term_to_binary({WorkerPid, Ref, catch erlang:apply(M, F, A)}),
+    Ret = case catch erlang:apply(M, F, A)of
+               {'EXIT', _} = V -> {badrpc, V};
```

Comment:
This should return badrpc for errors too! Look:

    rpc:call(node(), erlang, error, [test]).
    {badrpc,{'EXIT',{test,[{rpc,local_call,3,
                            [{file,"rpc.erl"},{line,330}]},
                       {erl_eval,do_apply,6,[{file,"erl_eval.erl"},{line,673}]},
                       {shell,exprs,7,[{file,"shell.erl"},{line,686}]},
                       {shell,eval_exprs,7,[{file,"shell.erl"},{line,641}]},
                       {shell,eval_loop,3,[{file,"shell.erl"},{line,626}]}]}}}